src/Control/Cookie.php

```diff
@@ -38,73 +32,63 @@ class Cookie
      *
      * @return Cookie_Backend
      */
-    public static function get_inst()
+    public static function get_inst(): Cookie_Backend
     {
-        return Injector::inst()->get('SilverStripe\\Control\\Cookie_Backend');
+        return Injector::inst()->get(Cookie_Backend::class);
     }
 
     /**
      * Set a cookie variable.
      *
      * Expiry time is set in days, and defaults to 90.
      *
-     * @param string $name
-     * @param mixed $value
-     * @param float $expiry
-     * @param string $path
-     * @param string $domain
-     * @param bool $secure
-     * @param bool $httpOnly
-     *
      * See http://php.net/set_session
      */
     public static function set(
-        $name,
-        $value,
-        $expiry = 90,
-        $path = null,
-        $domain = null,
-        $secure = false,
-        $httpOnly = true
+        string $name,
+        string|false $value,
+        int $expiry = 90,
+        ?string $path = null,
+        ?string $domain = null,
+        bool $secure = false,
+        bool $httpOnly = true,
+        string $sameSite = ''
```

Comment:
- `$value` can only be a string or `false` if you trace through the calls in the actual implementation.
- `$expiry` can only be `int`.
- Chose to keep the nullability of `$path` and `$domain` to reduce the change of unexpected side effects and regressions given we're so close to the beta.

Reply:
These types are shared with `CookieJar` and `Cookie_Backend` as well.